tools/extract-metadata.py

```diff
@@ -160,34 +161,47 @@ def main():
 
     tree = etree.Element("xep-infos")
 
+    has_error = False
+
     for xepfile in args.xepdir.glob("xep-*.xml"):
         number = xepfile.name.split("-", 1)[1].split(".", 1)[0]
         try:
             number = str(int(number))
         except ValueError:
             continue
 
-        with xepfile.open("rb") as f:
-            parsed = open_xml(f)
-
-        tree.append(make_metadata_element(
-            number,
-            extract_xep_metadata(parsed),
-            True,
-        ))
+        try:
+            with xepfile.open("rb") as f:
+                parsed = open_xml(f)
+
+            tree.append(make_metadata_element(
+                number,
+                extract_xep_metadata(parsed),
+                True,
+            ))
+        except Exception as exc:
```

Comment:
What exceptions can this throw and are they likely? I think in CI it would be nicer if this just raised the acception and crashed since it's probably only likely to happen if some OSException occures (which is actually exceptional). Having a `has_error` flow is confusing and feels brittle too.